models/import_ufo.py

```diff
@@ -2514,6 +2514,26 @@ def remove_interactions(self, zero_couplings):
                                  ' with loop particles (%s)'%loop_parts+\
                                  ' perturbing order %s'%order)  
 
+        # looping over all vertex and remove all link to lorentz structure that are not used anymore
```

Comment:
not sure to understand this chunk of code...

Reply:
So originally (before restriction) the vertices  are like this 
color = [c1,c2, ...] 
lorentz = [l1,l2,....]
coupling = {(i1,j1): C1, (i2,j2): C2, ....}

When running the restriction, some coupling are dropped from the model.
But before this patch, neither the lorentz list nor the color list were modified (so just the coupling list was made shorter).

So the issue (only for loop) is that those "not used" lorentz structure was kept in the comparison operator and therefore they were kept separated. 

So this code is cleaning the lorentz structure and changing the coupling part since the index are modified when I'm doing that.

Reply:
ok. is there any unittest checking this, just to be sure not to break it in future changes?

Reply:
good point. I was lazy on that.

They were already  a unittest related to that function, but obviously was not focussing on that point.
So I have added two unittest to test one "trivial" case and a second less trivial. 
By limiting myself to SM case (for speed), I do not  deeply test most complex case that can be generated but I think that this is enough.

I have also run those unittest on 2.9 version, and they indeed fail (as expected).